lib/posix/Kconfig

```diff
@@ -14,11 +14,43 @@ config	POSIX_MAX_FDS
 
 config POSIX_API
 	bool "POSIX APIs"
+	select POSIX_HEADERS
```

Comment:
I think Ulf would tell you to not use select in this way if you can avoid it.
Would a "default y if POSIX_API" in this option below be good enough?

Reply:
The purpose of that config option is to select a bunch of other options.

Reply:
Yes, I understand, but the keyword "select" has a meaning beyond the english meaning of select.
At least Ulf recommends to use select only to set promtless symbols when possible, as otherwise it can be very confusing.

Reply:
`select` might be alright as currently used here. Could turn messy if dependencies are added. The [Kconfig best practices](https://docs.zephyrproject.org/latest/guides/kconfig/index.html) page has this:

    Rare exceptions might include cases where you’re sure that the dependencies of the selecting
    and selected symbol will never drift out of sync, e.g. when dealing with two simple symbols defined
    close to one another within the same if.

I'd add this to the help text too: "Particular POSIX APIs can still be enabled separately when this option is disabled."